presto-common/src/main/java/com/facebook/presto/common/RuntimeStats.java

```diff
@@ -26,17 +27,18 @@
  */
 public class RuntimeStats
 {
-    private final Map<String, RuntimeMetric> metrics;
+    private final ConcurrentMap<String, RuntimeMetric> metrics;
 
     public RuntimeStats()
     {
-        metrics = new HashMap<>();
+        metrics = new ConcurrentHashMap<>();
     }
 
     @JsonCreator
     public RuntimeStats(Map<String, RuntimeMetric> metrics)
     {
-        this.metrics = requireNonNull(metrics, "metrics is null");
+        requireNonNull(metrics, "metrics is null");
+        this.metrics = new ConcurrentHashMap<>(metrics);
```

Comment:
these 2 can be merged into a single line -

```
this.metrics = new ConcurrentHashMap<>(metrics);
```

Reply:
Updated this to use metrics.forEach, in case metrics is modified during this copy.